src/gcp/cloudlogging.ts

```diff
@@ -0,0 +1,101 @@
+import * as api from "../api";
+import { FirebaseError } from "../error";
+import { logger } from "../logger";
+
+const API_VERSION = "v2";
+
+export interface LogEntry {
+  logName: string;
+  resource: any;
+  timestamp?: string;
+  receiveTimestamp: string;
+  severity?: any;
+  insertId?: string;
+  httpRequest?: any;
+  labels?: any;
+  metadata?: any;
+  operation?: any;
+  trace?: string;
+  spanId?: string;
+  traceSampled?: boolean;
+  sourceLocation?: any;
+  protoPayload?: any;
+  textPayload?: string;
+  jsonPayload?: any;
+}
+
+/**
+ * GCP api call to list all log entries (https://cloud.google.com/logging/docs/reference/v2/rest/v2/entries/list)
+ */
+export async function listEntries(
+  projectId: string,
+  filter: string,
+  pageSize: number,
+  order: string
+): Promise<LogEntry[]> {
+  const endpoint = `/${API_VERSION}/entries:list`;
+  try {
+    const result = await api.request("POST", endpoint, {
+      auth: true,
+      data: {
+        resourceNames: [`projects/${projectId}`],
+        filter: filter,
+        orderBy: "timestamp " + order,
+        pageSize: pageSize,
+      },
+      origin: api.cloudloggingOrigin,
+    });
+    return result.body.entries;
+  } catch (err) {
+    throw new FirebaseError("Failed to retrieve log entries from Google Cloud.", {
+      original: err,
+    });
+  }
+}
+
+/**
+ * The correct API filter to use when GCFv2 is enabled and/or we want specific function logs
+ * @param v2Enabled check if the user has the preview v2 enabled
+ * @param functionList list of functions seperated by comma
+ * @returns the correct filter for use when calling the list api
+ */
+export function getApiFilter(v2Enabled: boolean, functionList?: string) {
```

Comment:
You've put GCF-specific code in a cloudlogging library. Let's move this back to the functions logging code.

Reply:
Makes sense to remove from the gcf library, instead of putting the functions back into the `Command` where we can't export them, I think it we should move them into a file like `src/functions/functionsLog.ts`, thoughts?

Reply:
SGTM

Reply:
done